internal/core/core.go

```diff
@@ -720,17 +751,16 @@ func (cs *coreService) GetRoomMessages(
 		QueryParams: &queryParams,
 	})
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer response.Body.Close()
 
-	var messages []Message
-	err = common.DecodeResponseBody(response.Body, &messages)
+	err = common.DecodeResponseBody(response.Body, target)
```

Comment:
I did that initially but you have to do a bit of a dance to make it compile...

This won't compile:

```go
func f(*interface{}) {}

func main() {
	x := []int{}
	f(&x)
}
```

This will, but given the extra level of indirection... I'm not sure it's actually any clearer:

```go
func f(*interface{}) {}

func main() {
	x := []int{}
	y := interface{}(x)
	f(&y)
}
```

wdyt?

Reply:
scratch that, the second example doesn't even work, because we're mutating a copy of x... So yeah. I don't think you can have `*interface{}` do what you want it to.